### PR TITLE
rules: added check for the presence of a variable in the template

### LINKED ISSUE #754: Give a rule compile error if filters refer to the variables that are not present in the pattern
If pattern does not refer to `$x` we should probably give a rule compilation error:
```php
/**
 * @type int $x
 */
$_ = foo();
```

### PR DESCRIPTION
**Pull request by i582**
Fixes #754

### CODE REVIEW

File: src/ir/irutil/irtutil.go [hunk 1]
```diff
@@ -66,6 +66,29 @@ func FmtNode(n ir.Node) string {
 	return irfmt.Node(n)
 }
 
+// Find searches for a node in the passed subtree.
+func Find(what ir.Node, where ir.Node) bool {
```
Comment (quasilyte) [inline src/ir/irutil/irtutil.go:70]: Is this functions used?

Comment (i582) [reply src/ir/irutil/irtutil.go:70]: No, left for the future.

Comment (i582) [reply src/ir/irutil/irtutil.go:70]: By the way, we have already discussed that it is possible 
to collect information about variables and check them here, 
but I don’t remember exactly, but there were some problems 
that I could not solve.

Comment (quasilyte) [reply src/ir/irutil/irtutil.go:70]: That's not a good reason to keep unused code. :)

Comment (quasilyte) [reply src/ir/irutil/irtutil.go:70]: These util functions are almost never removed, even if unused. It's a good way to keep sanity with not adding more if you don't need them that much.

Comment (i582) [reply src/ir/irutil/irtutil.go]: Removed.

**Pull request merged**